SMLHelper/Options/ConfigModOptions.cs

```diff
@@ -0,0 +1,700 @@
+        /// <param name="e">The <see cref="ButtonClickedEventArgs"/> for the click event.</param>
+        private void ConfigModOptions_ButtonClicked(object sender, ButtonClickedEventArgs e)
+        {
+            if (modOptionsMetadata.TryGetValue(e.Id, out var modOptionMetadata)
+                && modOptionMetadata.MemberInfo is MethodInfo methodInfo)
+            {
+                var parameters = methodInfo.GetParameters();
+                var invokeParams = new object[parameters.Length];
+                var senderFound = false;
+                var eventArgsFound = false;
+
+                for (var i = 0; i < parameters.Length; i++)
+                {
+                    var param = parameters[i];
+                    if (!senderFound && param.ParameterType == typeof(object))
+                    {
+                        senderFound = true;
+                        invokeParams[i] = sender;
+                        continue;
+                    }
+
+                    if (!eventArgsFound && param.ParameterType == typeof(ButtonClickedEventArgs))
+                    {
+                        eventArgsFound = true;
+                        invokeParams[i] = e;
+                    }
+
+                    if (senderFound && eventArgsFound)
+                    {
+                        break;
+                    }
+                }
+
+                methodInfo.Invoke(Config, invokeParams);
+            }
+        }
+
+        /// <summary>
+        /// Sets the value in the <see cref="Config"/>, optionally saving the <see cref="Config"/> to disk if the
+        /// <see cref="MenuAttribute.SaveEvents.ChangeValue"/> flag is set, before passing off to
+        /// <see cref="InvokeOnChange{TEventArgs}(ModOptionMetadata, object, TEventArgs)"/> to invoke any methods specified with a
+        /// <see cref="OnChangeAttribute"/>.
+        /// </summary>
+        /// <param name="sender">The sender of the original choice changed event.</param>
+        /// <param name="e">The <see cref="ChoiceChangedEventArgs"/> for the choice changed event.</param>
+        private void ConfigModOptions_ChoiceChanged(object sender, ChoiceChangedEventArgs e)
+        {
+            if (modOptionsMetadata.TryGetValue(e.Id, out var modOptionMetadata))
+            {
+                // Set the value in the Config
+                switch (modOptionMetadata.MemberInfo)
+                {
+                    case PropertyInfo property when property.PropertyType.IsEnum && modOptionMetadata.ChoiceAttribute == null:
+                        property.SetValue(Config, Enum.Parse(property.PropertyType, e.Value), null);
+                        break;
+                    case PropertyInfo property when property.PropertyType.IsEnum:
+                        property.SetValue(Config,
+                            Enum.Parse(property.PropertyType, Enum.GetNames(property.PropertyType)[e.Index]), null);
```

Comment:
Either have all parameters of `SetValue` on their own lines (and aligned) or put them all on one line.

Reply:
I'm currently rewriting this (see my latest comment about how this is currently a **draft** bc I'm working on optimisations)

Reply:
Resolving now. Will look it over again after the rewrite.